moped-database/migrations/1719582711923_refine_funding_source/up.sql

```diff
@@ -0,0 +1,531 @@
+DROP VIEW IF EXISTS component_arcgis_online_view;
+DROP VIEW IF EXISTS project_list_view;
+
+CREATE OR REPLACE VIEW project_list_view AS WITH project_person_list_lookup AS (
+    SELECT
+        mpp.project_id,
+        string_agg(DISTINCT concat(mu.first_name, ' ', mu.last_name, ':', mpr.project_role_name), ','::text) AS project_team_members
+    FROM moped_proj_personnel AS mpp
+    INNER JOIN moped_users AS mu ON mpp.user_id = mu.user_id
+    INNER JOIN moped_proj_personnel_roles AS mppr ON mpp.project_personnel_id = mppr.project_personnel_id
+    INNER JOIN moped_project_roles AS mpr ON mppr.project_role_id = mpr.project_role_id
+    WHERE mpp.is_deleted = false AND mppr.is_deleted = false
+    GROUP BY mpp.project_id
+),
+
+funding_sources_lookup AS (
+    SELECT
+        mpf.project_id,
+        string_agg(DISTINCT mfs.funding_source_name, ', ' ORDER BY mfs.funding_source_name) AS funding_source_name,
+        string_agg(
+            DISTINCT
+            CASE
+                WHEN mfs.funding_source_name IS NOT null AND mfp.funding_program_name IS NOT null
+                    THEN concat(mfs.funding_source_name, ' - ', mfp.funding_program_name)
+                WHEN mfs.funding_source_name IS NOT null
+                    THEN mfs.funding_source_name
+                WHEN mfp.funding_program_name IS NOT null
+                    THEN mfp.funding_program_name
+            END,
+            ', '
+            ORDER BY
+                CASE
+                    WHEN mfs.funding_source_name IS NOT null AND mfp.funding_program_name IS NOT null
+                        THEN concat(mfs.funding_source_name, ' - ', mfp.funding_program_name)
+                    WHEN mfs.funding_source_name IS NOT null
+                        THEN mfs.funding_source_name
+                    WHEN mfp.funding_program_name IS NOT null
+                        THEN mfp.funding_program_name
+                END
```

Comment:
@mddilley I left three branches in here, where two of them -- the ones where it only has one of the two fields to display -- where two of them are broken out separately when they could be combined. My logic is that it leaves open an opportunity to add some sort of label to them, because as-is, it's not easy to tell if you're looking at a single program name or a single source name. LMK if you'd like these combined down into one.

Reply:
sounds great! I'll check back in on any refinements needed in my next check-in with Nathan in a couple weeks. Either way, I think the branching makes it easier to see what makes up this field. I've been trying to keep that in mind as we accumulate fields with more complex logic.